string type
uint32[] uids